Ember / Handlebars: Do not break opening mustache and path

dcyriller opened the pull request:
## Description

In Ember ecosystem, the opening mustache and the path are often printed on a same line (see [this comment](https://github.com/prettier/prettier/issues/8691#issuecomment-652589072) and [this comment](https://github.com/prettier/prettier/issues/8691#issuecomment-652557920) for more details).

### Changes

```handlebars
{{!-- Input --}}
<GlimmerComponent
  @errors={{or this.aVeryLongProperty (and this.aProperty (v-get bike "number" "message"))}}
  data-test-beneficiary-account-number
/>
<GlimmerComponent
  @progress={{aPropertyEngdingAfterEightiethColumnToHighlightAWeirdClosingParenIssue}}
/>

{{!-- Prettier stable --}}
<GlimmerComponent
  @errors={{
    or
    this.aVeryLongProperty
    (and this.aProperty (v-get bike "number" "message"))
  }}
  data-test-beneficiary-account-number
/>
<GlimmerComponent
  @progress={{
    aPropertyEngdingAfterEightiethColumnToHighlightAWeirdClosingParenIssue
  }}
/>

{{!-- Prettier main --}}
<GlimmerComponent
  @errors={{or
    this.aLongProperty
    (and this.aProperty (v-get bike "number" "message"))
  }}
  data-test-beneficiary-account-number
/>
<GlimmerComponent
  @progress={{aPropertyEngdingAfterEightiethColumnToHighlightAWeirdClosingParenIssue}}
/>
```

## Checklist

<!-- Please ensure you’ve done all of these things (if applicable). -->
<!-- You can replace the `[ ]` with `[x]` to mark each task as done. -->

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

<!-- Please DO NOT remove the playground link -->

**✨[Try the playground for this PR](https://deploy-preview-10586--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAeA4gGwJYFtdwBOAwhLgA7QIwA6UABPQAJGESEDOAvMMO-TAAW2DgDoAhgDUiATwAy0AOYAFNuSIwZ9ABTioAEwHCx41RHWFNOgG4BaRXBj0ARtgDWcejRBQArrmciby8QAg4OcQdvAEpogF84ukZ9cRhxW3gOGFtAqDgAM2wwbHFCGVtxMEhfWFs-AKCoAHoAPjoMHHwiUgoqWCTmcjZFQjhwnmBTNQ0ZAFEoRX1sBYBBfPhCWexFQRhsR0FSTH8oABUIAAltwRwdmBWAdThsQn1iTAgOZZVShABJcK+OAJOitEAAGhA5j20A4yFApTYAHdlL8oHCUOJMEjxDI4ZDnIRKh4YABlciVb7IGCEIGQuANfT6OD6OR6RS+SJwABi7FwqT2C2QIHEvhgEAhIF2uEwD2EmQpYDgpIQXz21mwmmFYHCkuWHA0qki-OQ+SxBshACsOAAPABCRLAJNJ4gIcmWcFN5rgVttpO+mDgAEVfBB4F7MBaQBTOERhYpOgRCJKhssYA9sPohMgABwABkhQ0+cAeRPIwqGYyI1k9kIAjqH4GZyBiRRw6nAWSzJaMGy84EbFCakGbIz6QAbcNhqbTx18FoGQ2HPSPvZC0s4M1nBMgAEzronYHALHrDkBjACskt8BtO4mcGNHUesQL+BmopLAhGw5HuBlJmiBhGBoJEAA)✨**

thorn0 commented:
Will this close #8691?

dcyriller commented:
It only covers a part of it

thorn0 commented:
Please add test cases from that issue. If you want, you can prefix the test files' names with `issue-8691` (e.g. `issue-8691-if-1.hbs`). I usually do that when I collect test cases from issues..

thorn0 commented on tests/handlebars/text-wrap/__snapshots__/jsfmt.spec.js.snap:
```diff
@@ -277,11 +277,7 @@ printWidth: 80
-  }}after{{stuff}}after{{stuff}}after{{stuff}}after{{stuff}}after{{
-    stuff
-  }}after{{stuff}}after{{stuff}}after{{stuff}}after
+  before{{stuff}}after{{stuff}}after{{stuff}}after{{stuff}}after{{stuff}}after{{stuff}}after{{stuff}}after{{stuff}}after{{stuff}}after{{stuff}}after{{stuff}}after{{stuff}}after{{stuff}}after{{stuff}}after
```
Is this intended?

dcyriller replied:
yes it is intended... was it prettier?

thorn0 replied:
I have no opinion on this. It just looked unrelated to the linked issue.

dcyriller replied:
I see, sorry for the confusion - I should have opened a specific issue for this PR

dcyriller commented:
> Please add test cases from that issue. If you want, you can prefix the test files' names with issue-8691 (e.g. issue-8691-if-1.hbs). I usually do that when I collect test cases from issues..

I have a follow-up PR ready for the linked issue. Do you mind if I apply this comment to the follow-up PR?

Indeed, I split the work in two PRs to have two different changelog entries. (So that it is clearer for the end user.) But if you prefer I can merge them. Let me know!

thorn0 commented:
> two different changelog entries. (So that it is clearer for the end user

This is a good idea. I meant only those test cases that this PR improves. Not all the test cases from that issue.

dcyriller commented:
I have added test cases taken from #8691 (it won't close the issue though) and rebased to highlight the difference (all commits are supposed to pass the CI).

thorn0 commented on changelog_unreleased/handlebars/10586.md:
```diff
@@ -0,0 +1,40 @@
+<GlimmerComponent
+  @errors={{or
+    this.aLongProperty
+    (and this.aProperty (v-get bike 'number' 'message'))
```
This doesn't match the input. Please fix.

thorn0 commented on changelog_unreleased/handlebars/10586.md:
```diff
@@ -0,0 +1,40 @@
+  @errors={{
+    or this.aLongProperty (and this.aProperty (v-get bike 'number' 'message'))
+    this.aLongProperty
+    (and this.aProperty (v-get bike 'number' 'message'))
```
This doesn't match the input. Please fix.

dcyriller replied:
sorry

thorn0 commented:
Wrapping the closing `}}` like this looks unexpected.

**Prettier pr-10586**
[Playground link](https://deploy-preview-10586--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAeA4gGwJYFtdwBOAwhLgA7QIwA6UABPQAJGESEDOAvMMO-TAAW2DgDoAhgDUiATwAy0AOYAFNuSIwZ9ABTioAEwHCx41RHWFNOgG4BaRXBj0ARtgDWcejRBQArrmciby8QAg4OcQdvAEpogF84ukZ9cRhxW3gOGFtAqDgAM2wwbHFCGVtxMEhfWFs-AKCoAHoAPjoQABoQcxhsaA5kUFK2AHdlUoQBlHFMEfEZAa7nQkqPGABlckrsKEVkGEJfOC64Bv19OH05PUVfSLgAMXZcVN7d5BBxXxgITpBBGC4TAAdWEmS2YDg60m2F61lhMg+YHCfx2HA0qkiL2Q+Rm6K6ACsOAAPABCKzAa3W4gIch2cBxeOOICJxPWO0UmDgAEVfBB4IzMPiQFtOEQPpy8ARCH9yIQdjBgdh9EJkABOAAMXTlEHRwJW5A+crg6MI1gZXQAjnz4GZyFNPhw6nBLpc-oQ4NbsB7MYpsUhcULmejcNh9odgxyubz+QyA0yumlnEqVYJkAAmRMrbA4XakfDiD4mgCsf186IAKuJnFNA8LrEcAJIGajrMDy8gwACCBnWmi5gvRCSAA)
<!-- prettier-ignore -->
```sh
--parser glimmer
--print-width 90
```

**Input:**
<!-- prettier-ignore -->
```hbs
<GlimmerComponent
  @errors={{or this.aVeryLongProperty (and this.aProperty (v-get bike "number" "message"))}}
  data-test-beneficiary-account-number
/>

```

**Output:**
<!-- prettier-ignore -->
```hbs
<GlimmerComponent
  @errors={{or this.aVeryLongProperty (and this.aProperty (v-get bike "number" "message"))
  }}
  data-test-beneficiary-account-number
/>
```

dcyriller commented:
nice catch - I pushed some changes (they specifically adress the issue reported in [this comment](https://github.com/prettier/prettier/issues/8691#issuecomment-731205444))
